### PR TITLE
Added simplified version of calculated APY 

### PR DESCRIPTION
**Pull request by Soniachniy**
And showing farm rewards for project validators

### CODE REVIEW

File: packages/frontend/src/components/common/balance/BalanceDisplay.js [hunk 1]
```diff
@@ -92,6 +96,8 @@ const BalanceDisplay = ({
         }
     };
 
+    const markIfNotNearAndNotWhitelisted = !isNear && !isWhiteListed ? '#FF585D' : '';
```
Comment (Soniachniy) [inline packages/frontend/src/components/common/balance/BalanceDisplay.js:99]: are there any constants for colors?

File: packages/frontend/src/components/staking/components/Validator.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Translate } from 'react-localize-redux';
 import { useDispatch } from 'react-redux';
 
 import { Mixpanel } from '../../../mixpanel';
 import { redirectTo } from '../../../redux/actions/account';
 import { actionsPending } from '../../../utils/alerts';
+import { PROJECT_VALIDATOR_VERSION } from '../../../utils/constants';
```
Comment (Soniachniy) [inline packages/frontend/src/components/staking/components/Validator.js]: will it be farming validator?

File: packages/frontend/src/components/staking/components/Validator.js [hunk 3]
```diff
@@ -57,6 +101,7 @@ export default function Validator({
                 <Translate id='staking.validator.button' />
             </FormButton>
             {validator && <StakingFee fee={validator.fee.percentage} />}
+            {validator?.version === PROJECT_VALIDATOR_VERSION && <StakingAPY apy={calculatedAPY} />}
```
Comment (Soniachniy) [inline packages/frontend/src/components/staking/components/Validator.js]: maybe we should make version not optional

File: packages/frontend/src/utils/ref-finance.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  } catch (error) {
+    console.error(`Failed to fetch token prices: ${error}`);
+  }
+};
+
+export const fetchTokenWhiteList = async (accountId) => {
+  try {
+    const account = wallet.getAccountBasic(accountId);
+    const contract = new Contract(account, REF_FINANCE_CONTRACT, {viewMethods: ['get_whitelisted_tokens']});
+    const whiteListedTokens = await contract.get_whitelisted_tokens({"from_index": 0,"limit": 100});
```
Comment (Soniachniy) [inline packages/frontend/src/utils/ref-finance.js]: get_whitelisted_tokens is not accepting params

File: packages/frontend/src/utils/staking.js [hunk 5]
```diff
@@ -93,3 +96,40 @@ export function shuffle(sourceArray) {
     }
     return sourceArray;
 }
+
+const SECONDS_IN_YEAR = 3600 * 24 * 365;
+const NEAR_TOKEN_ID = 'wrap.near';
```
Comment (Soniachniy) [inline packages/frontend/src/utils/staking.js]: process.env.NEAR_TOKEN_ENV || 'wrap.testnet'

File: packages/frontend/src/utils/staking.js [hunk 6]
```diff
@@ -93,3 +96,40 @@ export function shuffle(sourceArray) {
...
+        // TODO: check handle multiple farms, if some of token prices are unavailable.
+        if (activeFarms.some(farm => !+tokenPrices[farm.token_id]?.price)) return 0;
+
+        const totalStakedBalance = validator?.poolSummary.total_staked_balance;
+
+        const summaryAPY = activeFarms.reduce((acc, farm) => {
+            const tokenPriceInUSD = +tokenPrices[farm.token_id].price;
+            const nearPriceInUSD = +tokenPrices[NEAR_TOKEN_ID].price;
+
+            const rewardsPerSecond = farm.amount / (farm.end_date - farm.start_date) * 1e9;
```
Comment (Soniachniy) [inline packages/frontend/src/utils/staking.js]: should be done via Big or BN

File: packages/frontend/src/utils/staking.js [hunk 7]
```diff
@@ -93,3 +96,40 @@ export function shuffle(sourceArray) {
...
+            const rewardsPerSecondInUSD = rewardsPerSecond * tokenPriceInUSD;
```
Comment (Soniachniy) [inline packages/frontend/src/utils/staking.js]: All calculation should be with big numbers

File: packages/frontend/src/utils/staking.js [hunk 8]
```diff
@@ -93,3 +96,40 @@ export function shuffle(sourceArray) {
...
+            const totalStakedBalanceInUSD = totalStakedBalance * nearPriceInUSD;
+            const farmAPY = rewardsPerSecondInUSD * SECONDS_IN_YEAR / totalStakedBalanceInUSD * 100;
+            
+            return acc + farmAPY;
+        }, 0);
+        
+        // TODO: Check the rule of rounding for APY ;
+        const roundedAPY = Math.round(summaryAPY * 100000) / 100000;
```
Comment (Soniachniy) [inline packages/frontend/src/utils/staking.js]: to constants

Comment (Soniachniy) [reply packages/frontend/src/utils/staking.js]: can use utils from wallet

File: packages/frontend/src/components/common/balance/BalanceDisplay.js (hunk 1 above)
Comment (askulikov) [reply packages/frontend/src/components/common/balance/BalanceDisplay.js:99]: There isn't a constant for this color. Hex value used through the code instead

File: packages/frontend/src/components/staking/components/Validator.js (hunk 2 above)
Comment (askulikov) [reply packages/frontend/src/components/staking/components/Validator.js]: changed through the project

File: packages/frontend/src/components/staking/components/Validator.js (hunk 3 above)
Comment (askulikov) [reply packages/frontend/src/components/staking/components/Validator.js]: It has to be optional in this case, as validator might be undefined on page load

File: packages/frontend/src/utils/ref-finance.js (hunk 4 above)
Comment (askulikov) [reply packages/frontend/src/utils/ref-finance.js]: changed

File: packages/frontend/src/utils/staking.js (hunk 5 above)
Comment (askulikov) [reply packages/frontend/src/utils/staking.js]: put it in config file

File: packages/frontend/src/utils/staking.js (hunk 7 above)
Comment (askulikov) [reply packages/frontend/src/utils/staking.js]: Check please the correctness of calculations

File: packages/frontend/src/components/common/balance/BalanceDisplay.js (hunk 1 above)
Comment (Soniachniy) [reply packages/frontend/src/components/common/balance/BalanceDisplay.js:99]: we are using this one more than one, should be added somewhere?

File: packages/frontend/src/components/staking/components/Validator.js [hunk 9]
```diff
@@ -19,11 +23,18 @@ export default function Validator({
     loading,
     selectedValidator,
     currentValidators,
+    accountId,
 }) {
     const [confirm, setConfirm] = useState(null);
+    const currentValidator = useSelector(state => state.staking.currentValidator);
+    const isFarmingValidator = validator?.version === FARMING_VALIDATOR_VERSION;
```
Comment (Soniachniy) [inline packages/frontend/src/components/staking/components/Validator.js:30]: versions shouldn't be optional

File: packages/frontend/src/redux/actions/staking.js [hunk 10]
```diff
@@ -385,6 +370,7 @@ export const { staking } = createActions({
             ({ accountId, isOwner }) => ({ accountId, isOwner })
         ],
         GET_VALIDATORS: async (accountIds, accountId) => {
+            console.log('GET_VALIDATORS');
```
Comment (Soniachniy) [inline packages/frontend/src/redux/actions/staking.js:373]: console.log

**Pull request closed without merging**